ts/packages/anchor/src/coder/borsh/instruction.ts

```diff
@@ -303,11 +304,15 @@ class InstructionFormatter {
           const variants = typeDef.type.variants;
           const variant = Object.keys(data)[0];
           const enumType = data[variant];
+          const camelCaseVariant = camelCase(variant);
+          const enumVariant = variants.find((v) => camelCase(v.name) === camelCaseVariant);
+          const relevantFields = enumVariant?.fields;
           const namedFields = Object.keys(enumType)
             .map((f) => {
               const fieldData = enumType[f];
-              const idlField = variants[variant]?.find(
-                (v: IdlField) => v.name === f
+              const camelCaseField = camelCase(f);
+              const idlField = relevantFields?.find<IdlField>(
+                (v): v is IdlField => camelCase((v as IdlField).name) === camelCaseField
```

Comment:
Do you have an example code where the previous version fails?

This should be fixed in `0.29.0` https://github.com/coral-xyz/anchor/pull/2633 but it would fail if the IDL was generated using older `anchor-cli` version. I'm not sure whether we should support previous IDL versions but this change is small enough that it wouldn't hurt to have it.

Reply:
[Tx](https://solscan.io/tx/5ue4VoQZ6aucU2xDaLPnUFiYGh45TheEw3PUjtwnfRVsK941TxtMqnr5XQHbkBWCQwM9AeYK8WAUgyfdQP9j1ysw)
Try parsing #3 - Instruction 2 with this. [IDL](https://explorer.solana.com/address/JUP6LkbZbjS1jKKwapdHNy74zcZ3tLUZoi5QNyVTaV4/anchor-program) for quick access
Data hex: `0xc1209b3341d69c8105010000001b016400012b6ac827f8010000c25e3e2600000000320000`
Should roughly parse to
```js
{
  "accounts": [
    {
      "isSigner": false,
      "isWritable": false,
      "name": "Token Program",
      "pubkey": "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA"
    },
    {
      "isSigner": false,
      "isWritable": false,
      "name": "Program Authority",
      "pubkey": "CapuXNQoDviLvU1PxFiizLgPNQCxrsag1uMeyk6zLVps"
    },
    {
      "isSigner": true,
      "isWritable": true,
      "name": "User Transfer Authority",
      "pubkey": "DCAKxn5PFNN1mBREPWGdk1RXg5aVH9rPErLfBFEi2Emb"
    },
    {
      "isSigner": false,
      "isWritable": true,
      "name": "Source Token Account",
      "pubkey": "4symCyBEijtmLW88JKanqbR35UGUG2rqRU4P1wiD1kFH"
    },
    {
      "isSigner": false,
      "isWritable": true,
      "name": "Program Source Token Account",
      "pubkey": "56R4uBuaEdytagu1v9NdmUkZxGYBG7jGSG9iiLBufX43"
    },
    {
      "isSigner": false,
      "isWritable": true,
      "name": "Program Destination Token Account",
      "pubkey": "Gjmjory7TWKJXD2Jc6hKzAG991wWutFhtbXudzJqgx3p"
    },
    {
      "isSigner": false,
      "isWritable": true,
      "name": "Destination Token Account",
      "pubkey": "4twSoHoLny6443AnobaihFk2ch2ChKFbPcxaW4wp9orU"
    },
    {
      "isSigner": false,
      "isWritable": false,
      "name": "Source Mint",
      "pubkey": "MNDEFzGvMt87ueuHvVU9VcTqsAP5b3fTGPsHuuPA5ey"
    },
    {
      "isSigner": false,
      "isWritable": false,
      "name": "Destination Mint",
      "pubkey": "EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v"
    },
    {
      "isSigner": false,
      "isWritable": false,
      "name": "Platform Fee Account",
      "pubkey": "JUP6LkbZbjS1jKKwapdHNy74zcZ3tLUZoi5QNyVTaV4"
    },
    {
      "isSigner": false,
      "isWritable": false,
      "name": "Token2022 Program",
      "pubkey": "JUP6LkbZbjS1jKKwapdHNy74zcZ3tLUZoi5QNyVTaV4"
    },
    {
      "isSigner": false,
      "isWritable": false,
      "name": "Event Authority",
      "pubkey": "D8cy77BBepLMngZx6ZukaTff5hCt1HrWyKk3Hnd9oitf"
    },
    {
      "isSigner": false,
      "isWritable": false,
      "name": "Program",
      "pubkey": "JUP6LkbZbjS1jKKwapdHNy74zcZ3tLUZoi5QNyVTaV4"
    },
    {
      "isSigner": false,
      "isWritable": true,
      "pubkey": "CC9VYJprbxacpiS94tPJ1GyBhfvrLQbUiUSVMWvFohNW"
    },
    {
      "isSigner": false,
      "isWritable": true,
      "pubkey": "9kUhfVY8Re3ijCHHBDrPNi2ctfsLbyaa8hTris7WsygZ"
    },
    {
      "isSigner": false,
      "isWritable": true,
      "pubkey": "F8URBrjQm1HuHFCFDc6oqKMKnn5TLvwxeqEEMKdRW7Gw"
    },
    {
      "isSigner": false,
      "isWritable": true,
      "pubkey": "gAaxwxJFLQJ3XimtmGbRZjTa62tCF5y6xSDJRV2AQps"
    },
    {
      "isSigner": false,
      "isWritable": true,
      "pubkey": "5BbiqowpoEFiM3AF1aHo3txPeqfreuspbwCkeUxBnXzg"
    },
    {
      "isSigner": false,
      "isWritable": true,
      "pubkey": "8pjSudyoaEJSoe8kwWCETth7XxRcmxkDYNq7gexcCFL"
    },
    {
      "isSigner": false,
      "isWritable": true,
      "pubkey": "7gkt1TxEz8QwWbuzLLYwQX1GLJf6A96edojjnLkbXZjz"
    },
    {
      "isSigner": false,
      "isWritable": true,
      "pubkey": "3Gf8inHVdxpFqQCyK2gS4svmR6J72toNtGXnwNPCB6Bf"
    },
    {
      "isSigner": false,
      "isWritable": false,
      "pubkey": "BiRH4KoyUu65ATpXysV8USiuuezhqALqmKtUNHT9A34M"
    },
    {
      "isSigner": false,
      "isWritable": false,
      "pubkey": "CapuXNQoDviLvU1PxFiizLgPNQCxrsag1uMeyk6zLVps"
    },
    {
      "isSigner": false,
      "isWritable": true,
      "pubkey": "56R4uBuaEdytagu1v9NdmUkZxGYBG7jGSG9iiLBufX43"
    },
    {
      "isSigner": false,
      "isWritable": true,
      "pubkey": "56R4uBuaEdytagu1v9NdmUkZxGYBG7jGSG9iiLBufX43"
    },
    {
      "isSigner": false,
      "isWritable": true,
      "pubkey": "Gjmjory7TWKJXD2Jc6hKzAG991wWutFhtbXudzJqgx3p"
    },
    {
      "isSigner": false,
      "isWritable": false,
      "pubkey": "srmqPvymJeFKQ4zGQed1GFppgkRHL9kaELCbyksJtPX"
    },
    {
      "isSigner": false,
      "isWritable": false,
      "pubkey": "TokenkegQfeZyiNwAJbNbGKPFXCWuBvf9Ss623VQ5DA"
    },
    {
      "isSigner": false,
      "isWritable": false,
      "pubkey": "SysvarRent111111111111111111111111111111111"
    },
    {
      "isSigner": false,
      "isWritable": false,
      "pubkey": "JUP6LkbZbjS1jKKwapdHNy74zcZ3tLUZoi5QNyVTaV4"
    }
  ],
  "args": [
    {
      "data": "5",
      "name": "id",
      "type": "u8"
    },
    {
      "data": "[{ swap: Openbook { side: Ask }, percent: 100, inputIndex: 0, outputIndex: 1 }]",
      "name": "routePlan",
      "type": "Vec<RoutePlanStep>"
    },
    {
      "data": "2165330962987",
      "name": "inAmount",
      "type": "u64"
    },
    {
      "data": "641621698",
      "name": "quotedOutAmount",
      "type": "u64"
    },
    {
      "data": "50",
      "name": "slippageBps",
      "type": "u16"
    },
    {
      "data": "0",
      "name": "platformFeeBps",
      "type": "u8"
    }
  ],
  "contractName": "jupiter",
  "name": "sharedAccountsRoute"
}
```